src/qt/walletcontroller.cpp

```diff
@@ -147,23 +151,150 @@ void WalletController::removeAndDeleteWallet(WalletModel* wallet_model)
     delete wallet_model;
 }
 
+WalletControllerActivity::WalletControllerActivity(WalletController* wallet_controller, QWidget* parent_widget)
+    : QObject(wallet_controller)
+    , m_wallet_controller(wallet_controller)
+    , m_parent_widget(parent_widget)
+{
+}
+
+WalletControllerActivity::~WalletControllerActivity()
+{
+    delete m_progress_dialog;
+}
+
+void WalletControllerActivity::showProgressDialog(const QString& label_text)
+{
+    m_progress_dialog = new QProgressDialog(m_parent_widget);
+
+    m_progress_dialog->setLabelText(label_text);
+    m_progress_dialog->setRange(0, 0);
+    m_progress_dialog->setCancelButton(nullptr);
+    m_progress_dialog->setWindowModality(Qt::ApplicationModal);
+    GUIUtil::PolishProgressDialog(m_progress_dialog);
+
+    connect(m_progress_dialog, &QObject::destroyed, [this] {
+        m_progress_dialog = nullptr;
+    });
+}
+
+CreateWalletActivity::CreateWalletActivity(WalletController* wallet_controller, QWidget* parent_widget)
+    : WalletControllerActivity(wallet_controller, parent_widget)
+{
+    m_passphrase.reserve(MAX_PASSPHRASE_SIZE);
+}
+
+CreateWalletActivity::~CreateWalletActivity()
+{
+    delete m_create_wallet_dialog;
+    delete m_passphrase_dialog;
+}
 
-OpenWalletActivity::OpenWalletActivity(WalletController* wallet_controller, const std::string& name)
-    : m_wallet_controller(wallet_controller)
-    , m_name(name)
-{}
+void CreateWalletActivity::askPasshprase()
+{
+    m_passphrase_dialog = new AskPassphraseDialog(AskPassphraseDialog::Encrypt, m_parent_widget, &m_passphrase);
```

Comment:
A call to `m_passphrase_dialog->setWindowModality(Qt::ApplicationModal);` before `->show();` see to fix the autofocussing.

Reply:
Done

Reply:
Great, thanks.